### ISSUE TITLE
Accessibility: Double-click triggers 2 regular clicks.

### ISSUE THREAD

**Issue by thewaver** [google-map-react/google-map-react#953]
**Describe the bug 🐛**

Whenever I perform a double-click on a GoogleMapReact component I receive 2 "regular click" events.
The implication of this is that on double-click the map zooms AND performs the action that I have set for `onClick` twice - but I want (and expect) only the zoom to trigger.

**To Reproduce 🔍**

1. Create map component.
2. Set `onClick` callback.
3. Run project.
4. Double-click on map.
5. `onClick` callback fires twice.

**Expected behavior 💭**

I expect the `onClick` callback not to be called when I perform a double-click, given that the double-click is an independent interaction that triggers its own action (ie: zoom).

**Environment:**

 - OS: MacOS Catalina
 - Browser Chrome 84
 - Version 1.1.7
 - Using React Native 0.62 and TypeScript 3.9

**Comment by itsmichaeldiego** [comment]
@thewaver Thanks for submitting an issue, I think I know what you're talking about, but you're not being clear + you didn't provide a live example or a GIF at least of your issue.

What do you mean by `double-click on a GoogleMapReact component`, are you referring to a Marker, or what `GoogleMapReact component` are you referring to? 

Where did you set this `onClick`function you're referring to, to the Marker or to what? Its very hard to know without a code snippet also.

I will close this issue until we've the necessary information in order to help you

**Comment by itsmichaeldiego** [comment]
Thanks for submitting an issue, I think I know what you're talking about, but you're not being that clear. 
Also you didn't provide a live example / GIF / code snippet as we require!

What do you mean by `double-click on a GoogleMapReact component`? Are you referring to a Marker, or what `GoogleMapReact component` are you referring to? I can't guess from the info given. Also, where did you set this `onClick` function you're talking about? To the Marker or to what?? It is very hard to guess what you're clicking and what is the problem if we don't know where did you set this `onClick` function. 

Please be more thorough and thoughtful when submitting an issue, we're happy to help but we can't afford that extra guessing effort 😊

**Comment by thewaver** [comment]
Thank you for the response. I did not really expect that there would be any confusion here though.
According to "@types/google-map-react" your framework exports a single component called "GoogleMapReact" - so forgive me yet again, but I do not see where there could be any ambiguity regarding what I stated.

Regardless, here's a stitched-up example that requires you to still set proper keys and so forth but should work as a valid foundation:

```ts
import * as React from "react";
import GoogleMapReact, {ChangeEventValue, ClickEventValue} from "google-map-react";

const mapOptions: MapOptions = {
    disableDefaultUI: false,
    mapTypeControlOptions: ["satellite"],
    mapTypeControl: true
};

const MapLocationField = () => {
    const [lat, setLat] = React.useState("");
    const [lng, setLng] = React.useState("");

    const setNewCoords = React.useCallback(async (mapLat: number, mapLng: number) => {
        setLat(mapLat.toFixed(defaultDecimalPlaces));
        setLng(mapLng.toFixed(defaultDecimalPlaces));
    }, []);

    const handleMapClick = React.useCallback((eventValue: ClickEventValue) => {
        setNewCoords(eventValue.lat, eventValue.lng);
    }, [setNewCoords]);

    return (
        <GoogleMapReact
            options={mapOptions}
            bootstrapURLKeys={mapKeys}
            center={{lat: +lat, lng: +lng}}
            defaultZoom={17}
            zoom={17}
            onClick={handleMapClick}
        >
            <Marker key={Date.now()} lat={+lat} lng={+lng} />
        </GoogleMapReact>
    );
};
```

I basically trimmed my code down to this, so there may be syntax or other issues, but should be good enough to get the idea. (you probably need an extra container to set dimensions in there - I was using `Box` from Material UI)

A double-click anywhere on the map causes the situation that I described.

To remove any further ambiguity, the "map" is this:
![Screenshot 2020-09-28 at 14 27 04](https://user-images.githubusercontent.com/31538712/94426787-053c1080-0186-11eb-901d-0143581a7579.png)

**Comment by itsmichaeldiego** [comment]
@thewaver Hey man there was no need to get all passive-aggressive, I was just pointing out that its hard to help if we're not as descriptive as possible. 

Your gif is not moving, I still don't see what's wrong both google maps and our library work the same.

Here's a GIF of what you describe in [maps.google.com](https://www.google.com/maps):

![Screen Recording 2020-09-29 at 01 58 AM](https://user-images.githubusercontent.com/6419886/94497399-63550c00-01f7-11eb-8df7-a97141822506.gif)

**Comment by itsmichaeldiego** [comment]
After re-reading this, I think I understand what you're pointing out, I thought you were double-clicking on a Marker and it was zooming anyways. 

Anyways, without a live example (something like [codesandbox](http://codesandbox.io/)) we would have to create one and reproduce it, as it has never been reported, we do have some handling for onClick, this is the chunk of code that handles the click: [src/google_map.js#L956-L979](https://github.com/google-map-react/google-map-react/blob/master/src/google_map.js#L956-L979)

If you're having a bug and you really need it you could do so yourself and there're plenty of ways to do it, the most classic one is a timeout and a flag. 

This could be a good implementation on our side too, if its an issue I agree. If you're proposing that, please create a new issue with a live example and I could help implementing it, if a couple of more people want it I can go ahead and implement it myself too.

**Comment by thewaver** [comment]
From your own example, it appears that the behaviour is present in Google Maps, so your framework would not be at fault here. Hence I believe this can be closed. It is not so much that the issue is solved, but it would not be your fault or responsibility, and as such it does indeed appear that my options are limited to "debouncing the click" (not entirely sure that I am using the correct terminology here). IIRC, according to WCAG guidelines a double-click should not still trigger the effects of a single click (in case the double-click performs an action itself).

Keep up the good work.

**Comment by itsmichaeldiego** [comment]
@thewaver Thanks, it is weird indeed, I don't know what is the expected behavior, but I think it makes sense that it is triggered twice.

You could use a debounce yes, or you can use `event.detail`;

```javascript
function GoogleMapWrapper() {
  const handleClick = ({ event }) => {
    if (event.detail === 1) {
      console.log(`Click on Map`);
    }
  };

  return (
    <GoogleMap
      defaultZoom={10}
      defaultCenter={[34.0522, -118.2437]}
      style={{ height: "100%", width: "100%" }}
      onClick={handleClick}
    />
  );
}

```

source: https://stackoverflow.com/questions/880608/prevent-click-event-from-firing-when-dblclick-event-fires
